### PR TITLE
Add TemplateExtensions for markdown and asciidoc

### LINKED ISSUE #106: Add TemplateExtensions for asciidoc and markdown
`text.markdownify ` markdown to html
`text.asciidocify` asciidoc to html

### PR DESCRIPTION
**Pull request by mcruzdev**
Closes #106

### CODE REVIEW

**Review by ia3andy — APPROVED** [review]
LGTM

File: asciidoc/runtime/src/main/java/io/quarkiverse/qute/web/asciidoc/runtime/AsciidocSectionHelperFactory.java [hunk 1]
```diff
@@ -21,11 +29,24 @@ public List<String> getDefaultAliases() {
 
     @Override
     public AsciidocSectionHelper initialize(SectionInitContext context) {
-        return new AsciidocSectionHelper();
+        return new AsciidocSectionHelper(converter);
+    }
+
+    @Override
+    public CompletionStage<Object> resolve(EvalContext context) {
```
Comment (mkouba) [inline asciidoc/runtime/src/main/java/io/quarkiverse/qute/web/asciidoc/runtime/AsciidocSectionHelperFactory.java:36]: The disadvantage of this solution (a `ValueResolver` implementation) is that it cannot contribute to validation of type-safe templates, i.e. if there is a type-safe template with `{myString.asciidocify}` then we will not be able to validate this expression and depending on situation the build could fail and something like `quarkus.qute.type-check-excludes=*.asciidocify` would need to be used.

Therefore I recommend to use a regular [extension method](https://quarkus.io/guides/qute-reference#template_extension_methods) instead.

Something like this should work:

```java
@EngineConfiguration
public class AsciidocSectionHelperFactory {

   private static final AsciidocConverter converter = new AsciidocConverter();

    @TemplateExtension(matchNames = {"asciidocify", "toAsciidoc"}) 
    static String convertToAsciidoc(String text) { 
        return converter.apply(text);
    }
}
```

For me `asciidocify` sounds a bit like a tongue-twister. Therefore I recommend to add the `toAsciidoc` alias as well ;-).

And this comment also applies to the MD extension.

**Comment by mkouba** [comment]
@ia3andy @mcruzdev Pls check my updates. I'm sorry for the squash/force push without letting you know. Hope that's not a big problem.

**Comment by mkouba** [comment]
> > @ia3andy @mcruzdev Pls check my updates. I'm sorry for the squash/force push without letting you know. Hope that's not a big problem.
> 
> No worry @mkouba, was great to see how to solve this one. Why you added `LazyValue`? I would like to know to see how is the Qute build/runtime phase.

We added the lazy value so that the bean lookup is not performed for every invocation of the extension method at runtime.

**Comment by mkouba** [comment]
> Thank you for all! It help me a lot!

You're welcome ;-).

**Pull request merged**